libnetwork/sandbox.go

```diff
@@ -347,6 +348,15 @@ func (sb *Sandbox) GetEndpoint(id string) *Endpoint {
 	return nil
 }
 
+// RefreshResolverPolicy makes sure the resolver will only forward DNS requests to
+// upstream resolvers if proxying is enabled, and there's at least one endpoint with
+// external connectivity.
+func (sb *Sandbox) RefreshResolverPolicy() {
+	if sb.resolver != nil {
+		sb.resolver.SetForwardingPolicy(sb.getGatewayEndpoint() != nil)
```

Comment:
While it technically functions as intended given that `runtime.GOOS == "windows"` implies `sb.resolver == nil`, neither the `dnsNoProxy` flag nor `OptionDNSNoProxy` are in Windows-only code. As long as the option can be set on Linux sandboxes, the option should be effective.
```suggestion
		sb.resolver.SetForwardingPolicy(!sb.config.dnsNoProxy && sb.getGatewayEndpoint() != nil)
```
(Corollary: the option does not need to be effective on Linux sandboxes if it is a compile or runtime error to create a sandbox with `OptionDNSNoProxy()`.)

Reply:
`OptionDNSNoProxy()` now only exists on Windows.